### ISSUE TITLE
Merlin using incorrect ppx_jane executable

### ISSUE THREAD

**Issue by basus** [janestreet/jbuilder#199]
When using `jbuilder` generated `.merlin`-files, I get multiple errors of the form, inside Emacs:

```
(merlin) backtrace: 
Merlin failed with error: "Error while running external preprocessor
Command line: ../_build/default/.ppx/ppx_jane/ppx.exe --as-ppx --cookie 'library-name=conveyance' '/tmp/camlppx9378eb' '/tmp/camlppx47164d' 1>/dev/null 2>/dev/null"
```
Looking at the `.merlin` file, the relevant flag is `FLG -ppx "../_build/default/.ppx/ppx_jane/ppx.exe --as-ppx --cookie 'library-name="conveyance"'"`. I'm not sure why there is a `ppx_jane` executable under the `_build` directory, but when I use the `opam`-installed `ppx_jane` binary, the errors go away. I change the flag line to `FLG -ppx "$(opam config var bin)/ppx-jane --as-ppx --cookie 'library-name="conveyance"'"`.

`conveyance` is the name of my project. Right now, I'm manually editing the `.merlin` files by hand to change the `ppx_jane` binary. I can't tell if this is a bug with `jbuilder` or something wrong with my setup.

**Comment by bmillwood** [comment]
Perhaps this is related to #196 ?

**Comment by diml** [comment]
jbuilder generates custom ppx drivers on-demand in the `_build` directory, that's why the .merlin is pointing to the executable in the `_build` directory.

I'm not sure why it's failing though, maybe merlin is not running the ppx from where jbuilder expects. 

@let-def where is the the ppx run from by merlin?

**Comment by diml** [comment]
I talked with @trefis, there was indeed a related change in merlin and that's why the ppx binary is not found. I suppose in the short-term we can generate an absolute path in the .merlin. In the long term it might be problematic for ppx that read files, but hopefully there are not common.

**Comment by diml** [comment]
Should be fixed by #201
